src/GitHub.InlineReviews/ViewModels/InlineCommentPeekViewModel.cs

```diff
@@ -0,0 +1,204 @@
+using ReactiveUI;
+
+namespace GitHub.InlineReviews.ViewModels
+{
+    /// <summary>
+    /// Represents the contents of an inline comment peek view displayed in an editor.
+    /// </summary>
+    public sealed class InlineCommentPeekViewModel : ReactiveObject, IDisposable
+    {
+        readonly IApiClientFactory apiClientFactory;
+        readonly IInlineCommentPeekService peekService;
+        readonly IPeekSession peekSession;
+        readonly IPullRequestSessionManager sessionManager;
+        IPullRequestSession session;
+        IPullRequestSessionFile file;
+        IDisposable fileSubscription;
+        ICommentThreadViewModel thread;
+        IDisposable threadSubscription;
+        ITrackingPoint triggerPoint;
+        string fullPath;
+        bool leftBuffer;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="InlineCommentPeekViewModel"/> class.
+        /// </summary>
+        public InlineCommentPeekViewModel(
+            IApiClientFactory apiClientFactory,
+            IInlineCommentPeekService peekService,
+            IPeekSession peekSession,
+            IPullRequestSessionManager sessionManager,
+            INextInlineCommentCommand nextCommentCommand,
+            IPreviousInlineCommentCommand previousCommentCommand)
+        {
+            Guard.ArgumentNotNull(apiClientFactory, nameof(apiClientFactory));
+            Guard.ArgumentNotNull(peekService, nameof(peekService));
+            Guard.ArgumentNotNull(peekSession, nameof(peekSession));
+            Guard.ArgumentNotNull(sessionManager, nameof(sessionManager));
+            Guard.ArgumentNotNull(nextCommentCommand, nameof(nextCommentCommand));
+            Guard.ArgumentNotNull(previousCommentCommand, nameof(previousCommentCommand));
+
+            this.apiClientFactory = apiClientFactory;
+            this.peekService = peekService;
+            this.peekSession = peekSession;
+            this.sessionManager = sessionManager;
+            triggerPoint = peekSession.GetTriggerPoint(peekSession.TextView.TextBuffer);
+
+            peekSession.Dismissed += (s, e) => Dispose();
+
+            NextComment = ReactiveCommand.CreateAsyncTask(
+                Observable.Return(nextCommentCommand.IsEnabled),
+                _ => nextCommentCommand.Execute(new InlineCommentNavigationParams
+                {
+                    FromLine = peekService.GetLineNumber(peekSession, triggerPoint).Item1,
+                }));
+
+            PreviousComment = ReactiveCommand.CreateAsyncTask(
+                Observable.Return(previousCommentCommand.IsEnabled),
+                _ => previousCommentCommand.Execute(new InlineCommentNavigationParams
+                {
+                    FromLine = peekService.GetLineNumber(peekSession, triggerPoint).Item1,
+                }));
+        }
+
+        /// <summary>
+        /// Gets the thread of comments to display.
+        /// </summary>
+        public ICommentThreadViewModel Thread
```

Comment:
This should probably called something other than `Thread`, since there's a `Thread` type that can easily be confused with it, and makes the code harder to read. We call a series of related comments a `Discussion` on .com.